pkg/chunked/storage_linux.go

```diff
@@ -148,7 +164,26 @@ func makeZstdChunkedDiffer(ctx context.Context, store storage.Store, blobSize in
 	}, nil
 }
 
-func findFileInOtherLayers(file internal.ZstdFileMetadata, root string, dirfd int, layersMetadata map[string]map[string]*internal.ZstdFileMetadata, layersTarget map[string]string, missingDirsMode os.FileMode) (*os.File, int64, error) {
+func copyFileFromOtherLayer(file internal.ZstdFileMetadata, source string, otherFile *internal.ZstdFileMetadata, root string, dirfd int, missingDirsMode os.FileMode, useHardLinks bool) (bool, *os.File, int64, error) {
+	srcDirfd, err := unix.Open(source, unix.O_RDONLY, 0)
+	if err != nil {
+		return false, nil, 0, err
+	}
+	defer unix.Close(srcDirfd)
+
+	srcFile, err := openFileUnderRoot(otherFile.Name, source, srcDirfd, unix.O_RDONLY, 0)
+	if err != nil {
+		return false, nil, 0, err
+	}
+	defer srcFile.Close()
+
+	srcPath := fmt.Sprintf("/proc/self/fd/%d", srcFile.Fd())
+
+	dstFile, written, err := copyFileContent(srcPath, file.Name, root, dirfd, missingDirsMode, 0, useHardLinks)
+	return true, dstFile, written, err
+}
+
+func findFileInOtherLayers(file internal.ZstdFileMetadata, root string, dirfd int, layersMetadata map[string]map[string]*internal.ZstdFileMetadata, layersTarget map[string]string, missingDirsMode os.FileMode, useHardLinks bool) (bool, *os.File, int64, error) {
```

Comment:
Documenting the parameters and return values would be nice.

---

(`missingDirMode` seems unused, except passing it to functions that don’t use it.)